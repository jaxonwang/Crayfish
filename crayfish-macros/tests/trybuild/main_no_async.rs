#[crayfish_macros::main]
pub fn foo() -> Result<()>{
    println!("hello foo");
    Ok(())
} 
