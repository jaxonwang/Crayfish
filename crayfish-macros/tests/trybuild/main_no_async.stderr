error: Crayfish requires main function to be 'async'
 --> $DIR/main_no_async.rs:2:5
  |
2 | pub fn foo() -> Result<()>{
  |     ^^^^^^^^^^^^^^^^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/main_no_async.rs:1:1
  |
1 | / #[crayfish_macros::main]
2 | | pub fn foo() -> Result<()>{
3 | |     println!("hello foo");
4 | |     Ok(())
5 | | }
  | |_^ consider adding a `main` function to `$DIR/tests/trybuild/main_no_async.rs`
